application/contract/rootdomain/rootdomain.go

```diff
@@ -30,139 +29,139 @@ import (
-			fmt.Errorf("[ getUserInfoMap ] Can't get implementation: %s", err.Error())
+var INSATTR_Info_API = true
+
+// Info returns information about basic objects
+func (rd RootDomain) Info() (interface{}, error) {
+	migrationDaemonsMembersOut := []string{}
+	for _, ref := range rd.MigrationDaemonMembers {
+		migrationDaemonsMembersOut = append(migrationDaemonsMembersOut, ref.String())
 	}
 
-	balance, err := w.GetBalance()
+	res := map[string]interface{}{
+		"rootDomain":             rd.GetReference().String(),
+		"rootMember":             rd.RootMember.String(),
+		"migrationDaemonMembers": migrationDaemonsMembersOut,
+		"migrationAdminMember":   rd.MigrationAdminMember.String(),
+		"nodeDomain":             rd.NodeDomain.String(),
+	}
+	resJSON, err := json.Marshal(res)
 	if err != nil {
-		return map[string]interface{}{"member": name},
-			fmt.Errorf("[ getUserInfoMap ] Can't get total balance: %s", err.Error())
+		return nil, fmt.Errorf("failed to marshal result: %s", err.Error())
 	}
-	return map[string]interface{}{
-		"member": name,
-		"wallet": balance,
-	}, nil
+	return resJSON, nil
 }
 
-// DumpUserInfo processes dump user info request
-func (rd *RootDomain) DumpUserInfo(reference string) ([]byte, error) {
-	caller := *rd.GetContext().Caller
-	ref, err := insolar.NewReferenceFromBase58(reference)
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpUserInfo ] Failed to parse reference: %s", err.Error())
-	}
-	if *ref != caller && caller != rd.RootMember {
-		return nil, fmt.Errorf("[ DumpUserInfo ] You can dump only yourself")
-	}
-	m := member.GetObject(*ref)
+func (rd *RootDomain) AddBurnAddresses(burnAddresses []string) error {
+	rd.FreeBurnAddresses = append(rd.FreeBurnAddresses, burnAddresses...)
 
-	res, err := rd.getUserInfoMap(m)
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpUserInfo ] Problem with making request: %s", err.Error())
-	}
+	return nil
+}
 
-	return json.Marshal(res)
+func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
+	rd.FreeBurnAddresses = append(rd.FreeBurnAddresses, burnAddress)
+
+	return nil
 }
 
-// DumpAllUsers processes dump all users request
-func (rd *RootDomain) DumpAllUsers() ([]byte, error) {
-	if *rd.GetContext().Caller != rd.RootMember {
-		return nil, fmt.Errorf("[ DumpAllUsers ] Only root can call this method")
+func (rd RootDomain) GetBurnAddress() (string, error) {
+	if len(rd.FreeBurnAddresses) == 0 {
+		return "", fmt.Errorf("no more burn address left")
 	}
-	res := []map[string]interface{}{}
-	iterator, err := rd.NewChildrenTypedIterator(member.GetPrototype())
-	if err != nil {
-		return nil, fmt.Errorf("[ DumpAllUsers ] Can't get children: %s", err.Error())
+
+	return rd.FreeBurnAddresses[0], nil
+}
+
+func (rd *RootDomain) AddNewMemberToMaps(publicKey string, burnAddress string, memberRef insolar.Reference) error {
+	if _, ok := rd.PublicKeyMap[TrimPublicKey(publicKey)]; ok {
+		return fmt.Errorf("member for this publicKey already exist")
```

Comment:
It would be nice to have test for this case (we had one for members with same names, its practically the same)